### DISCUSSION TITLE
Keep the code exclude the extension initialization

### DISCUSSION THREAD

**Discussion by lucian-torje-siemens** [eclipse-edc/Connector#1926] [General]
Hello,
What is the best way to keep the code from an extension (reuse) but make sure that the initialization is not run (does not register beans).
So far I managed to do it using a customer extension loader and simple exclude the unneeded/to be overwritten extension 

![image](https://user-images.githubusercontent.com/102717569/189145439-33f4d0b0-209c-4c25-841e-0ea448f4dca1.png)

**Comment by jimmarino** [comment]
Doing something like that is brittle and error-prone. The best way to handle this is to keep the extension in question off the classpath.

**Reply by lucian-torje-siemens** [reply]
Thank you @jimmarino

**Reply by jimmarino** [reply]
What are you trying to do that requires the IDS extensions to be replaced? Are you trying to create a data sharing protocol as a substitute for IDS?

**Reply by lucian-torje-siemens** [reply]
I need to do some extra processing before IDS sends the data - this is not the best solution but it is a solution that works end-to-end and can be improved further. 

The main issue is that I could not find a good way to transfer the thread context info from the controller thread into the state machine thread running ProviderContractNegotiationManagerImpl and TransferProcessManagerImpl processing (and using opentelemetry context does not seem right). What would it the best way to accomplish this?
